### PR TITLE
Add nightly GUI test cases for OIDC authentication scenario

### PR DESCRIPTION
**Pull request by danfengliu**
OIDC authentication is a new feature in version 1.8.0, so add nightly GUI test cases for OIDC authentication scenario.

Signed-off-by: danfengliu <danfengl@vmware.com>

### CODE REVIEW

File: tests/robot-cases/Group1-Nightly/OIDC.robot [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Test Case - Create An New Project
+    Sign In Harbor With Ldap User    ${HARBOR_URL}
+    ${d}=    Get Current Date    result_format=%M%S
+    Create An New Project  test${d}
+    Close Browser
+
+Test Case - Delete A Project
+    Init Chrome Driver
+    Sign In Harbor With Ldap User    ${HARBOR_URL}
+    ${json}=  Run Curl And Return Json  curl -s -k -X GET --header 'Accept: application/json' -u '${HARBOR_ADMIN}:${HARBOR_PASSWORD}' 'https://10.192.123.153/api/users/search?username=${OIDC_USERNAME}'
```
Comment (wy65701436) [inline tests/robot-cases/Group1-Nightly/OIDC.robot]: why hard code the IP address? it's should be an internal one, don't expose to the public world.

File: tests/resources/Harbor-Pages/OIDC_Auth.robot [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    Init Chrome Driver
+    Go To    ${url}
+    Retry Element Click    ${log_oidc_provider_btn}
+    Retry Text Input    ${dex_login_btn}    ${username}@example.com
+    Retry Text Input    ${dex_pwd_btn}    password
+    Retry Element Click    ${submit_login_btn}
+    Retry Element Click    ${grant_btn}
+
+    ${isVisible}=  Run Keyword And Return Status  Element Should Be Visible  ${oidc_username_input}
+    Run Keyword If  '${isVisible}' == 'True'  Run Keywords  Retry Text Input    ${oidc_username_input}    ${username}  AND  Retry Element Click    ${save_btn}
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/OIDC_Auth.robot]: can we separate it into several steps? It's hard to understand.

File: tests/resources/Harbor-Pages/OIDC_Auth.robot [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+Documentation  This resource provides any keywords related to the Harbor private registry appliance
+Resource  ../../resources/Util.robot
+
+*** Variables ***
+${HARBOR_VERSION}  v1.1.1
+
+*** Keywords ***
+
+Sign In Harbor With Ldap User
+    [Arguments]  ${url}  ${username}=${OIDC_USERNAME}
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/OIDC_Auth.robot]: add some comments for the sign in workflow? Someone may not have a background about it.

File: tests/robot-cases/Group1-Nightly/OIDC.robot [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+// limitations under the License.
+
+*** Settings ***
+Documentation  Harbor BATs
+Resource  ../../resources/Util.robot
+Default Tags  Nightly
+
+*** Variables ***
+${HARBOR_URL}  https://${OIDC_HOSTNAME}
+${SSH_USER}  root
```
Comment (wy65701436) [inline tests/robot-cases/Group1-Nightly/OIDC.robot]: I don't get where to use this variable.

**Comment by wy65701436** [comment]
And, this PR should cherry-pick to release-1.8.0

File: tests/robot-cases/Group1-Nightly/OIDC.robot [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    Sign In Harbor With Ldap User    ${HARBOR_URL}    test7
+    Sleep  2
+    Sign In Harbor With Ldap User    ${HARBOR_URL}    test8
+    Sleep  2
+    Sign In Harbor With Ldap User    ${HARBOR_URL}    test9
+    Sleep  2
+    Close Browser
+
+Test Case - Create An New Project
+    Sign In Harbor With Ldap User    ${HARBOR_URL}
```
[the remark this answers is not in the document]
Comment (danfengliu) [reply tests/robot-cases/Group1-Nightly/OIDC.robot]: User name is made up in keyword of Sign In Harbor With OIDC User, like ${username}@example.com which is test1@example.com.

**Pull request merged**